piptools/build.py

```diff
@@ -41,6 +48,67 @@ class ProjectMetadata:
     build_requirements: tuple[InstallRequirement, ...]
 
 
+def maybe_statically_parse_project_metadata(
+    src_file: pathlib.Path,
+) -> ProjectMetadata | None:
```

Comment:
Given that we're using a static type checker, I think that would actually be less safe. mypy will complain if you do not handle None, but it will not complain if you do not handle the exception